src/domains/integration/controller/integration.controller.ts

```diff
@@ -1,33 +1,43 @@
 import { type PendingMemberMailsRepository, PendingMemberMailsRepositoryImpl } from 'domains/pendingMemberMail/repository';
 import { type OrganizationRepository, OrganizationRepositoryImpl } from '@domains/organization/repository';
 import { type IntegrationService, IntegrationServiceImpl } from '@domains/integration/service';
-import { ProjectIdIntegrationInputDTO } from '@domains/integration/dto';
+import { ProjectIdIntegrationInputDTO, type ProjectPreIntegratedDTO } from '@domains/integration/dto';
+import { type IssueProviderRepository, IssueProviderRepositoryImpl } from '@domains/issueProvider/repository';
+import { type ProjectManagementToolAdapter } from '@domains/adapter/projectManagementToolAdapter';
+import { type PendingProjectAuthorizationRepository, PendingProjectAuthorizationRepositoryImpl } from '@domains/pendingProjectAuthorization/repository';
 require('express-async-errors');
 
 export const integrationRouter: Router = Router();
 
 const projectRepository: ProjectRepository = new ProjectRepositoryImpl(db);
 const userRepository: UserRepository = new UserRepositoryImpl(db);
 const pendingAuthRepository: PendingProjectAuthorizationRepository = new PendingProjectAuthorizationRepositoryImpl(db);
-const projectTool: ProjectManagementToolAdapter = new LinearAdapter();
+const adapter: ProjectManagementToolAdapter = new LinearAdapter();
 const pendingMemberMailsRepository: PendingMemberMailsRepository = new PendingMemberMailsRepositoryImpl(db);
 const organizationRepository: OrganizationRepository = new OrganizationRepositoryImpl(db);
-const service: IntegrationService = new IntegrationServiceImpl(projectTool, projectRepository, userRepository, pendingAuthRepository, pendingMemberMailsRepository, organizationRepository);
+const issueProviderRepository: IssueProviderRepository = new IssueProviderRepositoryImpl(db);
+const service: IntegrationService = new IntegrationServiceImpl(adapter, projectRepository, userRepository, pendingAuthRepository, pendingMemberMailsRepository, organizationRepository, issueProviderRepository);
 
-integrationRouter.post('/project/linear', validateRequest(ProjectIdIntegrationInputDTO, 'body'), async (req: Request<any, any, ProjectIdIntegrationInputDTO>, res: Response) => {
-  const { sub } = res.locals.context as CustomCognitoIdTokenPayload;
+integrationRouter.post('/linear/project', validateRequest(ProjectIdIntegrationInputDTO, 'body'), async (req: Request<any, any, ProjectIdIntegrationInputDTO>, res: Response) => {
   const { projectId } = req.body;
 
-  const project: ProjectDTO = await service.integrateProject(projectId, sub);
+  const project: ProjectDTO = await service.integrateProject(projectId);
 
   res.status(HttpStatus.CREATED).json(project);
 });
+
+integrationRouter.get('/linear/projects', withAwsAuth, validateRequest(ProviderKeyDTO, 'query'), async (req: Request, res: Response): Promise<void> => {
```

Comment:
Why do you use aws middleware here? At this point, access token is already validated

Reply:
When we integrate project after authorization we receive an email with the confirmation where token access from aws is not provided. I removed middleware from the router and decided to use it in the endpoints needed